libsolidity/formal/CHC.cpp

```diff
@@ -643,19 +644,19 @@ set<Expression const*, CHC::IdCompare> CHC::transactionAssertions(ASTNode const*
 
 vector<VariableDeclaration const*> CHC::stateVariablesIncludingInheritedAndPrivate(ContractDefinition const& _contract)
 {
-	vector<VariableDeclaration const*> stateVars;
-	for (auto const& contract: _contract.annotation().linearizedBaseContracts)
-		for (auto var: contract->stateVariables())
-			stateVars.push_back(var);
-	return stateVars;
+	return fold(
+		_contract.annotation().linearizedBaseContracts,
+		vector<VariableDeclaration const*>{},
+		[](auto&& _acc, auto _contract) { return _acc + _contract->stateVariables(); }
```

Comment:
@chriseth from the other thread, this would then have to look like
`[](vector<VariableDeclaration const*>&& _acc, ContractDefinition const*) { return ...; }`
so the lambda is fully defined and `T` can be derived then

Reply:
I think I'd prefer requiring the initial accumulator. This also makes sense because not always the initial accumulator is `{}` (multiplication eg)

Reply:
ok!